### DISCUSSION TITLE
Return a custom (html) page _without_ going to the backend service

### DISCUSSION THREAD

**Discussion by msavy** [apiman/apiman#1744] [Help] [answered]
I received this interesting question directly and thought it was worth answering and reposting for everyone's benefit:

> Is it possible to have the Apiman plugin return a custom (html) page _without_ going to the backend service at all?

Yes, this is possible!

### Implement [`IConnectorInterceptor`](https://github.com/apiman/apiman/blob/master/gateway/engine/core/src/main/java/io/apiman/gateway/engine/policy/IConnectorInterceptor.java)

This lets you override the real backend connector and return whatever information you want to the requestor.

This could be just headers, but there's nothing stopping you also responding with body chunks.

### Set your interceptor with `context.setConnectorInterceptor`

In your custom policy, set your custom interceptor using `context#setConnectorInterceptor`

See an example from the Apiman CORS policy here: https://github.com/apiman/apiman-plugins/blob/master/cors-policy/src/main/java/io/apiman/plugins/cors_policy/CorsPolicy.java#L71-L78

Depending on the specifics of your policy's logic, you may want to call `#doSkip(request)` as in the example above.

Thanks for this question, Duncan / @dabloem!  Let us know how it goes.

**Comment by msavy** [comment, marked as answer]
Question answered above :-)

**Reply by volkflo** [reply]
Interesting idea, would be cool to have such a plugin at some point :)

**Reply by msavy** [reply]
@dabloem is writing a policy relating to Bitcoin LSAT protocol and sent me a message on Twitter asking a few details.

Check it out

https://twitter.com/dabloem/status/1477792509224960000

**Reply by dabloem** [reply]
@volkflo @msavy updated draft version. see https://github.com/dabloem/apiman-lsat-policy#readme
your feedback is most welcome

**Reply by msavy** [reply]
From a very brief scan it looks good!

Only thing I can see off the top of my head is that you've accidentally left the copyright headers from whatever you used as the example.
